File: Algorithm/src/Algorithm/Python/programmers/kakao/P67256.py
def find_func(keypad, i):
    for x in range(4):
        for y in range(3):
            if keypad[x][y] == i:
                return [x, y]


def solution(numbers, hand):
    answer = ''
    keypad = [
        [1, 2, 3],
        [4, 5, 6],
        [7, 8, 9],
        ['*', 0, '#']
    ]
    left = [3, 0]
    right = [3, 2]
    for i in numbers:
        if i in (3, 6, 9):
            answer += 'R'
            right = find_func(keypad, i)
            print(right)
        elif i in (1, 4, 7):
            answer += 'L'
            left = find_func(keypad, i)
            print(left)
        else:
            now = find_func(keypad, i)
            left_comparison = abs(now[0] - left[0]) + abs(now[1] - left[1])
            right_comparison = abs(now[0] - right[0]) + abs(now[1] - right[1])
            if left_comparison > right_comparison:
                right = now
                answer += 'R'
            elif left_comparison < right_comparison:
                left = now
                answer += 'L'
            else:
                if hand == 'right':
                    right = now
                    answer += 'R'
                else:
                    left = now
                    answer += 'L'
    return answer


numbers2 = [7, 0, 8, 2, 8, 3, 1, 5, 7, 6, 2]
hand2 = "left"
print(solution(numbers2, hand2))

File: Algorithm/src/Algorithm/Java/programmers/kakao/P67256.java
package Algorithm.Java.programmers.kakao;

import java.util.*;

public class P67256 {


    public static void main(String[] args) {
        System.out.println(solution(new int[]{1, 3, 4, 5, 8, 2, 1, 4, 5, 9, 5},"right"));
    }

    public static String solution(int[] numbers, String hand) {
        int[] posL = {3,0};
        int[] posR = {3,2};
        StringBuilder answer = new StringBuilder();
        String[] numberArr = Arrays.stream(numbers).mapToObj(String::valueOf).toArray(String[]::new);
        String[][] arr = new String[][]{{"1","2","3"},{"4","5","6"},{"7","8","9"},{"*","0","#"}};
        String[] L = {"1", "4", "7"};
        String[] R = {"3", "6", "9"};
        for (int i = 0; i < numberArr.length; i++) {
            String val = numberArr[i];
            if (val.equals("1") || val.equals("4") || val.equals("7")) {
                answer.append("L");
                for (int j = 0; j < L.length; j++) {
                    if (val.equals(L[j])) {
                        posL[0] = j;
                        posL[1] = 0;
                        break;
                    }
                }
            }
            else if (val.equals("3") || val.equals("6") || val.equals("9")) {
                answer.append("R");
                for (int j = 0; j < R.length; j++) {
                    if (val.equals(R[j])) {
                        posR[0] = j;
                        posR[1] = 2;
                        break;
                    }
                }
            } else {
                for (int j = 0; j < 4; j++){
                    if (val.equals(arr[j][1])) {
                        int lDistance = Math.abs(posL[0] - j) + Math.abs(posL[1] - 1);
                        int rDistance = Math.abs(posR[0] - j) + Math.abs(posR[1] - 1);
                        if (lDistance > rDistance) {
                            posR[0] = j;
                            posR[1] = 1;
                            answer.append("R");
                        } else if (rDistance > lDistance) {
                            posL[0] = j;
                            posL[1] = 1;
                            answer.append("L");
                        } else {
                            if (hand.equals("right")) {
                                posR[0] = j;
                                posR[1] = 1;
                                answer.append("R");
                            } else {
                                posL[0] = j;
                                posL[1] = 1;
                                answer.append("L");
                            }
                        }
                        break;
                    }
                }
            }
        }
        return answer.toString();
    }
}
